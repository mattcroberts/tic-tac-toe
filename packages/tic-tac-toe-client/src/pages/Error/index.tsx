import * as React from "react";
import style from "./Error.css";

interface IProps {
    message: string;
}
export default class ErrorPage extends React.Component<
    IProps,
    { hasError: boolean }
> {
    private static getDerivedStateFromError(error: Error) {
        return { hasError: true, error };
    }

    constructor(props: IProps) {
        super(props);
        this.state = { hasError: false };
    }

    public componentDidCatch(error: Error, info: any) {
        console.log(error, info);
    }

    public render() {
        if (this.state.hasError || this.props.message) {
            return (
                <div className={style.root}>
                    <h1>Unfortunately there has been an error</h1>
                    {this.props.message ? (
                        <pre>{this.props.message}</pre>
                    ) : null}
                </div>
            );
        }

        return this.props.children;
    }
}
