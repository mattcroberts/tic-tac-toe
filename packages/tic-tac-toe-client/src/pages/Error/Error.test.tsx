import * as React from "react";
import { shallow } from "enzyme";

import Error from "./index";

describe("Winner", () => {
    const defaultProps = {
        message: "test error message"
    };

    it("should render with message", () => {
        const component = shallow(<Error {...defaultProps} />);

        expect(component).toMatchSnapshot();
    });
});
