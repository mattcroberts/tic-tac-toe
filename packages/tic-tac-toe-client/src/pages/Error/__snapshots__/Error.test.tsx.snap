// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner should render with message 1`] = `
<div
  className="root"
>
  <h1>
    Unfortunately there has been an error
  </h1>
  <pre>
    test error message
  </pre>
</div>
`;
